gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/routes/checkpermissions/CheckPermissions.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.gateway.service.endpoints.routes.checkpermissions;
+
+import org.eclipse.ditto.base.model.headers.DittoHeaders;
+import org.eclipse.ditto.base.model.json.JsonParsableCommand;
+import org.eclipse.ditto.json.JsonField;
+import org.eclipse.ditto.json.JsonFieldDefinition;
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonObjectBuilder;
+
+import java.util.LinkedHashMap;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.eclipse.ditto.base.api.common.CommonCommand;
+import org.eclipse.ditto.base.model.json.JsonSchemaVersion;
+import org.eclipse.ditto.json.JsonFactory;
+
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Command to check multiple permissions for different resources in a single request.
+ *
+ * @since 3.7.0
+ */
+@Immutable
+@JsonParsableCommand(typePrefix = CheckPermissions.TYPE_PREFIX, name = CheckPermissions.NAME)
+public final class CheckPermissions extends CommonCommand<CheckPermissions> {
+
+    static final String TYPE_PREFIX = CommonCommand.TYPE_PREFIX;
+
+    /**
+     * The name of the command.
+     */
+    static final String NAME = "checkPermissions";
+
+    /**
+     * The type of the command.
+     */
+    public static final String TYPE = TYPE_PREFIX + CheckPermissions.NAME;
+
+    private static final JsonFieldDefinition<JsonObject> PERMISSION_CHECKS_FIELD = JsonFactory.newJsonObjectFieldDefinition(
+            "permissionChecks"
+    );
+
+    private final LinkedHashMap<String, ImmutablePermissionCheck> permissionChecks;
+
+    /**
+     * Constructs a new {@code CheckPermissionsCommand} object.
+     *
+     * @param dittoHeaders the headers of the command.
+     * @param permissionChecks a linked hash map of permission checks to be performed.
+     */
+    private CheckPermissions(final DittoHeaders dittoHeaders,
+            final LinkedHashMap<String, ImmutablePermissionCheck> permissionChecks) {
+        super(TYPE, Category.QUERY, dittoHeaders);
+        this.permissionChecks = new LinkedHashMap<>(permissionChecks);
```

Comment:
```suggestion
        this.permissionChecks = Collections.unmodifiableMap(new LinkedHashMap<>(permissionChecks));
```

Need to make this immutable, otherwise the class itself is not immutable.